packages/web/src/components/nav/desktop/AccountSwitcher/AccountSwitcher.tsx

```diff
@@ -49,6 +48,34 @@ export const AccountSwitcher = () => {
     return managedAccounts.filter(({ grant }) => grant.is_approved)
   }, [managedAccounts])
 
+  // Reset to the web3User if we detect that the current user is no longer in
+  // the managed accounts list
+  useEffect(() => {
+    if (
+      !currentUserId ||
+      !currentWeb3User ||
+      accountsStatus !== Status.SUCCESS ||
+      checkedAccess
+    ) {
+      return
+    }
+    // Check this only once per mount
+    setCheckedAccess(true)
```

Comment:
It's actually here because if you remove yourself as a manager in the settings, this effect runs so quickly that we don't have time to submit the EntityManager transaction to actually perform the removal (due to the optimistic update).
This is mostly just to keep folks infinitely in a state where things don't work even across refreshes.